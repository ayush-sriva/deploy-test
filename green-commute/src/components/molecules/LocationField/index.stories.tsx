import Box from '@mui/material/Box';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ChangeEvent } from 'react';
import LocationField from '.';

export default {
  title: 'components/Molecules/LocationField',
  component: LocationField,
} as ComponentMeta<typeof LocationField>;

const Template: ComponentStory<typeof LocationField> = (args) => (
  <Box sx={{ backgroundColor: '#0B6D62' }}>
    <LocationField {...args} />
  </Box>
);

export const FilledLocationField = Template.bind({});

FilledLocationField.args = {
  location: 'East Maredpally, Secunderabad',
  width: '350',
  onChangeHandler: (event: ChangeEvent<HTMLInputElement>) => {
    console.log('Hello');
  },
};
