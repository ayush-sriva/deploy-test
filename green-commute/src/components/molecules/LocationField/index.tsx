/**
 * @author Abhishek Bhemisetty <abhishek.bhemisetty@zemosolabs.com>
 */

import InputAdornment from '@mui/material/InputAdornment';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import CustomSvgIcon from '../../atoms/IconSVG';
import { ReactComponent as LocationIcon } from '../../../../public/assets/icons/locationIconWhite.svg';
import theme from '../../../theme';
import React, { ChangeEvent } from 'react';
import { makeStyles } from '@mui/styles';
import { styled } from '@mui/material/styles';

const useStyles = makeStyles({
  input: {
    color: `${theme.palette.primary.contrastText} !important`,
  },
});

interface LocationFieldProps {
  width: string;
  location: string;
  onChangeHandler: (event: ChangeEvent<HTMLInputElement>) => void;
}

const CssTextField = styled(TextField)({
  width: '320px',
  height: '36px',
  paddingBottom: `${theme.spacing(2)}`,
  '& .MuiInputBase-input': {
    color: `${theme.palette.primary.contrastText}`,
    '&:hover fieldset': {
      borderColor: `${theme.palette.primary.contrastText}`,
    },
  },
  '& .MuiInput-underline:before': {
    borderBottom: '2px solid white !important',
    transition: 'none ',
  },
  '& .MuiInput-underline:hover': {
    borderColor: `${theme.palette.primary.contrastText}`,
    transition: 'none ',
  },
  '& .MuiInput-underline:after': {
    borderBottomColor: 'white',
    transition: 'none ',
  },
});

const LocationBox: React.FC<LocationFieldProps> = ({ width, location, onChangeHandler, ...props }) => {
  const classes = useStyles();

  const wrapHandleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    onChangeHandler(event);
  };

  return (
    <Box>
      <CssTextField
        id='input-with-icon-textfield'
        data-testid='textfield-loc'
        InputProps={{
          startAdornment: (
            <InputAdornment position='start'>
              <CustomSvgIcon component={LocationIcon} htmlColor='transparent' />
            </InputAdornment>
          ),
          inputProps: { className: classes.input },
        }}
        variant='standard'
        defaultValue={location}
        onChange={wrapHandleChange}
        {...props}
      />
    </Box>
  );
};

export default LocationBox;
