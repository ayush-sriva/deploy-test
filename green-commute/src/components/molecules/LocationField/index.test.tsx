import { render, screen } from '@testing-library/react';
import ReactTestRenderer from 'react-test-renderer';
import LocationField from '.';

describe('Location Field Molecule', () => {
  test('Random location', () => {
    render(<LocationField location='Random Location' width='250px' onChangeHandler={() => {}}></LocationField>);
    expect(screen.getByTestId('textfield-loc')).toBeInTheDocument();
  });
});

describe('Location Field Snapshot', () => {
  it('renders location field correctly', () => {
    const tree = ReactTestRenderer.create(
      <LocationField location='Test Location' width='250px' onChangeHandler={() => {}}></LocationField>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
