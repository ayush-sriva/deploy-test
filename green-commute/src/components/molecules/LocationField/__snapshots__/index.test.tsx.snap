// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Field Snapshot renders location field correctly 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-ozplz9-MuiFormControl-root-MuiTextField-root"
    data-testid="textfield-loc"
  >
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1ptx2yq-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          color="transparent"
          data-testid="SVGIcon"
          focusable="false"
        >
          locationIconWhite.svg
        </svg>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart makeStyles-input-2 css-1x51dt5-MuiInputBase-input-MuiInput-input"
        defaultValue="Test Location"
        disabled={false}
        id="input-with-icon-textfield"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;
